### ISSUE TITLE
[0.6.2] Unexpected error adding highlightings/tokens on file

### ISSUE THREAD

**Issue by hientranvn2014** [gretard/sonar-tsql-plugin#11]
Hi gretard,

After upgrading to version 0.6.2, I saw a lot of these errors as below:

```
12:44:40.000 DEBUG: Unexpected error adding highlightings/tokens on file for token 'MONEY' on (43, 61) -  (43, 74)
java.lang.IllegalArgumentException: Unable to highlight file
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:97)
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:37)
	at org.sonar.plugins.tsql.sensors.antlr4.AntlrHighlighter.work(AntlrHighlighter.java:51)
	at org.sonar.plugins.tsql.sensors.HighlightingSensor.execute(HighlightingSensor.java:90)
```

And it took around 15 minutes to analyse 369 files => very slow. 
`12:43:09.732 INFO: Sensor HighlightingSensor [tsql] (done) | time=793447ms`

The tool versions that I'm using:
SonarQube version: 6.7
sonar-tsql-plugin: 0.6.2
SQLCodeGuard: 2.9

Please help me
Thanks

**Comment by gretard** [comment]
Hi,

Thank you for raising this. I was running tests with smaller amount of files. The thing is that custom rules were added to the plugin, which you can disable by setting property sonar.tsql.skip.custom.rules to true as they might be time consuming. And for errors - T-SQL grammar was updated, so that's why those errors might be shown. I will double check if code is correct for code highlighting. Also, I would suggest to disable DEBUG mode if possible :)

Greta

**Comment by hientranvn2014** [comment]
Hi Greta,
I turned off custom rules as you said and the build time decreased so much(around 5 minutes). Is there something wrong in checking custom rules? In the future I maybe need to use them.

About DEBUG mode, I only turn on it when needing to see detail errors.

Thank for your support
Larry

**Comment by gretard** [comment]
Hi,

I have updated how custom rules are handled. From my testing on large project (562 files) time has decreased from around 300s to 150s. However, still the improvement might not be such big as performance testing is not very easy :)

Greta
